node/network/availability-recovery/src/lib.rs

```diff
@@ -774,18 +742,20 @@ impl AvailabilityRecoverySubsystem {
 				_v = awaited_chunk_cleanup_interval.next() => {
 					cleanup_awaited_chunks(&mut state);
 				}
-				v = state.connection_requests.next().fuse() => {
-					if let Err(e) = handle_validator_connected(
-						&mut state,
-						&mut ctx,
-						v.validator_id,
-						v.peer_id,
-					).await {
-						tracing::warn!(
-							target: LOG_TARGET,
-							err = ?e,
-							"Failed to handle a newly connected validator",
-						);
+				v = state.connecting_validators.next() => {
+					if let Some((StreamYield::Item(v), _)) = v {
```

Comment:
It seems that we should match on `StreamYield::Finished` and invoke `remove` on that, otherwise the stream will be kept alive internally. 

https://docs.rs/streamunordered/0.5.1/src/streamunordered/lib.rs.html#650

Default behavior is `keep`: https://docs.rs/streamunordered/0.5.1/src/streamunordered/lib.rs.html#672 .